File: app/src/main/java/se/jakob/knarkklocka/viewmodels/AlarmActivityViewModel.java
package se.jakob.knarkklocka.viewmodels;

import android.app.Application;

public class AlarmActivityViewModel extends AlarmViewModel {
    public AlarmActivityViewModel(Application application) {
        super(application);
    }

    public void setAlarm(long id) {
        super.mAlarm = mRepository.getLiveAlarmByID(id);
    }
}

File: app/src/main/java/se/jakob/knarkklocka/viewmodels/AlarmActivityViewModel.kt
package se.jakob.knarkklocka.viewmodels

import se.jakob.knarkklocka.data.AlarmRepository

class AlarmActivityViewModel(repository: AlarmRepository, id: Long) : AlarmViewModel(repository) {

    override var liveAlarm = repository.getLiveAlarmByID(id)

}
